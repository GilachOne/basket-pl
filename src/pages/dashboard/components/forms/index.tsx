import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import {
    Main,
    FormDiv,
    Form,
    InputDiv,
    CreateListLabel,
    Input,
    CancelButton,
    CreateButton,
    ButtonDiv,
} from './style';

type CreateListProps = {
    listName?: string;
    onCancel: () => void;
    onOk: (listName: string) => void;
};

export const CreateListComponent: React.FC<CreateListProps> = ({
    listName,
    onCancel,
    onOk,
}) => {
    const { t } = useTranslation();
    const [inputValue, setInputValue] = useState(listName);

    const onSubmit = event => {
        event.preventDefault();
        inputValue
            ? onOk(inputValue)
            : onOk(t('basket.dashboard.createlistform.placeholder'));
    };

    const onChange = event => {
        setInputValue(event.target.value);
    };

    return (
        <Main>
            <FormDiv>
                <Form onSubmit={onSubmit}>
                    <InputDiv>
                        <CreateListLabel>
                            {t('basket.dashboard.createlistform.listname')}
                        </CreateListLabel>
                        <Input
                            type="text"
                            value={inputValue}
                            onChange={onChange}
                            placeholder={t(
                                'basket.dashboard.createlistform.placeholder'
                            )}
                        />
                    </InputDiv>
                    <ButtonDiv>
                        <CancelButton
                            onClick={onCancel}
                            title={t(
                                'basket.dashboard.createlistform.cancelbutton'
                            )}
                        >
                            {t('basket.dashboard.createlistform.cancelbutton')}
                        </CancelButton>
                        {listName ? (
                            <CreateButton
                                type="submit"
                                title={t(
                                    'basket.dashboard.deletelistform.okbutton'
                                )}
                            >
                                {t('basket.dashboard.deletelistform.okbutton')}
                            </CreateButton>
                        ) : (
                            <CreateButton
                                type="submit"
                                title={t(
                                    'basket.dashboard.createlistform.createbutton'
                                )}
                            >
                                {t(
                                    'basket.dashboard.createlistform.createbutton'
                                )}
                            </CreateButton>
                        )}
                    </ButtonDiv>
                </Form>
            </FormDiv>
        </Main>
    );
};

CreateListComponent.defaultProps = { listName: '' };

type DeleteListProps = {
    onCancel: () => void;
    onOk: () => void;
};

export const DeleteListComponent: React.FC<DeleteListProps> = ({
    onCancel,
    onOk,
}) => {
    const { t } = useTranslation();

    const OnSubmit = event => {
        event.preventDefault();
        onOk();
    };

    return (
        <Main>
            <FormDiv>
                <Form onSubmit={OnSubmit}>
                    <InputDiv>
                        <CreateListLabel>
                            {t('basket.dashboard.deletelistform.question')}
                        </CreateListLabel>
                    </InputDiv>
                    <ButtonDiv>
                        <CancelButton
                            onClick={onCancel}
                            title={t(
                                'basket.dashboard.createlistform.cancelbutton'
                            )}
                        >
                            {t('basket.dashboard.createlistform.cancelbutton')}
                        </CancelButton>
                        <CreateButton
                            type="submit"
                            title={t(
                                'basket.dashboard.deletelistform.okbutton'
                            )}
                        >
                            {t('basket.dashboard.deletelistform.okbutton')}
                        </CreateButton>
                    </ButtonDiv>
                </Form>
            </FormDiv>
        </Main>
    );
};
