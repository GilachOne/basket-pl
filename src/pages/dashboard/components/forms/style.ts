import styled from '@emotion/styled';

export const Main = styled.main`
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: ${props => props.theme.colors.accent.light};
    position: fixed; /* or absolute */
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
`;

export const FormDiv = styled.div`
    display: flex;
    flex-direction: column;
    background-color: ${props => props.theme.colors.text.main};
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    padding: 20px;
    margin: 10px;
`;

export const Form = styled.form``;

export const InputDiv = styled.p``;

export const CreateListLabel = styled.label`
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    display: flex;
    color: ${props => props.theme.colors.background.main};
`;

export const Input = styled.input`
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 300;
    font-size: 18px;
    display: flex;
    border: none;
    border-bottom: 1px solid ${props => props.theme.colors.text.light};
    width: 100%;
    height: 52px;
    :focus {
        outline: none;
    }
`;

export const ButtonDiv = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
`;

export const CancelButton = styled.button`
    cursor: pointer;
    padding: 15px;
    background: ${props => props.theme.colors.text.main};
    border: 1px solid ${props => props.theme.colors.background.main};
    border-radius: 3px;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    line-height: 21px;
    display: flex;
    color: ${props => props.theme.colors.background.main};
    width: 164px;
    height: 52px;
    justify-content: center;
    margin-right: 10px;
`;

export const CreateButton = styled.button`
    cursor: pointer;
    padding: 15px;
    background: ${props => props.theme.colors.background.main};
    border-radius: 3px;
    border: 0px;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    line-height: 21px;
    display: flex;
    color: ${props => props.theme.colors.text.main};
    width: 164px;
    height: 52px;
    justify-content: center;
    margin-left: 10px;
`;
