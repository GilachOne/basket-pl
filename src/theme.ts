export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
