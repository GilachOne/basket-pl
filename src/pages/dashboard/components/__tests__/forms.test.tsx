import React from 'react';
import { describe, it, expect, jest } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { CreateListComponent, DeleteListComponent } from '../forms/index';
import { theme } from '../../../../theme';

describe('Create, Rename and Delete Forms', () => {
    it('Create List', () => {
        const onCancel = jest.fn();
        const onOk = jest.fn();

        render(
            <ThemeProvider theme={theme}>
                <CreateListComponent onCancel={onCancel} onOk={onOk} />
            </ThemeProvider>
        );

        fireEvent.click(screen.getByTitle('Отмена'));
        fireEvent.click(screen.getByTitle('Создать'));
        fireEvent.change(screen.getByPlaceholderText('Новый список'), {
            target: { value: '23' },
        });

        expect(onCancel).toBeCalledTimes(1);
        expect(onOk).toBeCalled();
        expect(document.body).toMatchSnapshot();
    });

    it('Rename List', () => {
        const onCancel = jest.fn();
        const onOk = jest.fn();
        render(
            <ThemeProvider theme={theme}>
                <CreateListComponent
                    listName="Первый список"
                    onCancel={onCancel}
                    onOk={onOk}
                />
            </ThemeProvider>
        );

        fireEvent.click(screen.getByTitle('Отмена'));
        fireEvent.click(screen.getByTitle('OK'));

        expect(onCancel).toBeCalledTimes(1);
        expect(onOk).toBeCalled();
        expect(document.body).toMatchSnapshot();
    });

    it('Delete List', () => {
        const onCancel = jest.fn();
        const onOk = jest.fn();
        render(
            <ThemeProvider theme={theme}>
                <DeleteListComponent onCancel={onCancel} onOk={onOk} />
            </ThemeProvider>
        );

        fireEvent.click(screen.getByTitle('Отмена'));
        fireEvent.click(screen.getByTitle('OK'));

        expect(onCancel).toBeCalledTimes(1);
        expect(onOk).toBeCalled();
        expect(document.body).toMatchSnapshot();
    });
});
