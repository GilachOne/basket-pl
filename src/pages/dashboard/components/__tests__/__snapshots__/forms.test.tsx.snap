// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create, Rename and Delete Forms Create List 1`] = `
<body>
  <div>
    <main
      class="css-1yfhmvi"
    >
      <div
        class="css-1qehna1"
      >
        <form
          class="css-0"
        >
          <p
            class="css-0"
          >
            <label
              class="css-w4k18f"
            >
              Имя списка
            </label>
            <input
              class="css-ga4orh"
              placeholder="Новый список"
              type="text"
              value="23"
            />
          </p>
          <div
            class="css-alwjd5"
          >
            <button
              class="css-i6gg3r"
              title="Отмена"
            >
              Отмена
            </button>
            <button
              class="css-er65dn"
              title="Создать"
              type="submit"
            >
              Создать
            </button>
          </div>
        </form>
      </div>
    </main>
  </div>
</body>
`;

exports[`Create, Rename and Delete Forms Delete List 1`] = `
<body>
  <div>
    <main
      class="css-1yfhmvi"
    >
      <div
        class="css-1qehna1"
      >
        <form
          class="css-0"
        >
          <p
            class="css-0"
          >
            <label
              class="css-w4k18f"
            >
              Удалить список ?
            </label>
          </p>
          <div
            class="css-alwjd5"
          >
            <button
              class="css-i6gg3r"
              title="Отмена"
            >
              Отмена
            </button>
            <button
              class="css-er65dn"
              title="OK"
              type="submit"
            >
              OK
            </button>
          </div>
        </form>
      </div>
    </main>
  </div>
</body>
`;

exports[`Create, Rename and Delete Forms Rename List 1`] = `
<body>
  <div>
    <main
      class="css-1yfhmvi"
    >
      <div
        class="css-1qehna1"
      >
        <form
          class="css-0"
        >
          <p
            class="css-0"
          >
            <label
              class="css-w4k18f"
            >
              Имя списка
            </label>
            <input
              class="css-ga4orh"
              placeholder="Новый список"
              type="text"
              value="Первый список"
            />
          </p>
          <div
            class="css-alwjd5"
          >
            <button
              class="css-i6gg3r"
              title="Отмена"
            >
              Отмена
            </button>
            <button
              class="css-er65dn"
              title="OK"
              type="submit"
            >
              OK
            </button>
          </div>
        </form>
      </div>
    </main>
  </div>
</body>
`;
